### PR TITLE
Settings rework

### PR DESCRIPTION
**Pull request by DavidPetkovsek**

### CODE REVIEW

File: src/Atlas.cc [hunk 1]
```diff
@@ -82,23 +82,23 @@ void Atlas::AddMapPoint(MapPoint* pMP) {
...
     if (pCam->GetType() == GeometricCamera::CAM_PINHOLE) {
-      if (std::reinterpret_pointer_cast<Pinhole>(pCam_i)->IsEqual(pCam)) {
+      if (pCam_i->IsEqual(pCam)) {
         bAlreadyInMap = true;
         index_cam = i;
       }
     } else if (pCam->GetType() == GeometricCamera::CAM_FISHEYE) {
-      if (std::reinterpret_pointer_cast<KannalaBrandt8>(pCam_i)->IsEqual(pCam)) {
+      if (pCam_i->IsEqual(pCam)) {
         bAlreadyInMap = true;
```
Comment (DavidPetkovsek) [inline src/Atlas.cc:102]: did we change htis incorrectly?

File: src/Frame.cc [hunk 2]
```diff
@@ -423,10 +423,10 @@ Frame::Frame(const Camera_ptr &cam, const cv::Mat &imGray, const double &timeSta
...
                              static_cast<float>(mnMaxY - mnMinY);
 
-    fx = std::reinterpret_pointer_cast<Pinhole>(mpCamera)->toK().at<float>(0, 0);
-    fy = std::reinterpret_pointer_cast<Pinhole>(mpCamera)->toK().at<float>(1, 1);
-    cx = std::reinterpret_pointer_cast<Pinhole>(mpCamera)->toK().at<float>(0, 2);
-    cy = std::reinterpret_pointer_cast<Pinhole>(mpCamera)->toK().at<float>(1, 2);
+    fx = mpCamera->toK().at<float>(0, 0);
+    fy = mpCamera->toK().at<float>(1, 1);
+    cx = mpCamera->toK().at<float>(0, 2);
+    cy = mpCamera->toK().at<float>(1, 2);
```
Comment (DavidPetkovsek) [inline src/Frame.cc:429]: More reinterpret cast?

File: src/Frame.cc [hunk 3]
```diff
@@ -1233,7 +1231,7 @@ void Frame::ComputeStereoFishEyeMatches() {
           sigma1 = mvLevelSigma2[mvKeys[(*it)[0].queryIdx + monoLeft].octave],
           sigma2 =
               mvLevelSigma2[mvKeysRight[(*it)[0].trainIdx + monoRight].octave];
-      float depth = std::reinterpret_pointer_cast<KannalaBrandt8>(mpCamera)->TriangulateMatches(
+      float depth = std::static_pointer_cast<const KannalaBrandt8>(mpCamera)->TriangulateMatches(
```
Comment (DavidPetkovsek) [inline src/Frame.cc:1234]: reinterpret cast change

File: include/MORB_SLAM/Settings.h [hunk 4]
```diff
@@ -68,15 +63,19 @@ class Settings {
...
+
+    return distortionMat;
   }
-  cv::Mat camera2DistortionCoef() {
-    return cv::Mat(vPinHoleDistorsion2_.size(), 1, CV_32F,
-                   vPinHoleDistorsion2_.data());
+  cv::Mat camera2DistortionCoef() const {
+    cv::Mat distortionMat(vPinHoleDistorsion2_.size(), 1, CV_32F);
+    std::copy(vPinHoleDistorsion2_.begin(), vPinHoleDistorsion2_.end(), distortionMat.ptr<float>());
+
```
Comment (DavidPetkovsek) [inline include/MORB_SLAM/Settings.h:77]: ?

**Review by basedball — APPROVED** [review]

**Pull request merged**